bijection-core/src/main/scala/com/twitter/bijection/StringInjections.scala

```diff
@@ -32,9 +33,22 @@ trait StringInjections extends NumericInjections {
 
   def withEncoding(encoding: String): Injection[String, Array[Byte]] =
     new AbstractInjection[String, Array[Byte]] {
+      private[this] val decRef = {
+        val cs = Charset.forName(encoding)
+        new AtomicSharedState(() => cs.newDecoder.onUnmappableCharacter(CodingErrorAction.REPORT))
```

Comment:
is the charset threadsafe itself? 

Maybe we should have more like a () => {
val cs = Char...
})
? if threadsafe though doesn't seem like any reason to ofc